backend/app.js

```diff
@@ -63,4 +63,41 @@ app.post('/profile', (req, res) => {
     res.status(200).send("Profile endpoint under construction.");
 });
 
+// eventually read off session instead of userId in params
+app.put('/reset-password/:userId', async (req, res) => {
+    try {
+        const { userId } = req.params;
+        const { newPassword } = req.body;
+
+        if (userId == ':userId') {
+            console.log("Empty userId param.");
+            return res.status(400).send("Please log in first.");
+        };
+
+        const userInfo = await prisma.user.findUnique({
+            where: { id: userId },
+        });
+        
+        if (!userInfo) {
+            console.log("No user found.");
+            return res.status(500).send("Cannot find existing user.");
+        };
+
+        const saltRounds = 10;
+        const salt = await bcrypt.genSalt(saltRounds);
+        const hashedPassword = await bcrypt.hash(newPassword, salt);
+
+        await prisma.user.update({
+            where: { id: userId },
+            data: { password: hashedPassword },
+        });
+
+        res.send("Password Reset Successful.");
```

Comment:
Send back status 200 to indicate a successful put request, by default is it always set to 200 but it's better to write it out explicitly imo
```suggestion
        res.status(200).send("Password Reset Successful.");
```

Optional: Personally, I would send back a JSON object instead of a string as it offers better structure:

```js
res.status(200).json({
    message: "Successfully updated user password",
    success: true
});
```